File: Reserver/Util/FieldHelper.cs
namespace Reserver.Util;

public class FieldHelper
{
    public static IEnumerable<string> GetCategories(string? categories) =>
        ParseToIntArray(categories).Select(FieldHelper.GetCategoryString);


    public static IEnumerable<string> GetOffers(string? offers) =>
        ParseToIntArray(offers).Select(FieldHelper.GetOfferString);

    private static string GetOfferString(int id)
    {
        return id switch
        {
            0 => Offers.Unavailable,
            1 => Offers.Breakfast,
            2 => Offers.Brunch,
            3 => Offers.Lunch,
            4 => Offers.Supper,
            _ => Offers.Unavailable
        };
    }


    private static string GetCategoryString(int id)
    {
        return id switch
        {
            1 => Categories.Creative,
            2 => Categories.Pub,
            3 => Categories.Local,
            4 => Categories.Restaurant,
            5 => Categories.GoodTables,
            6 => Categories.Bistro,
            7 => Categories.National,
            8 => Categories.Breweries,
            9 => Categories.FamilyOwned,
            10 => Categories.FastFood,
            _ => Categories.Restaurant,
        };
    }

    private static IEnumerable<int> ParseToIntArray(string? jsonString) =>
        jsonString?.Split(',').Select(int.Parse).ToList() ?? Enumerable.Empty<int>();
}

internal struct Offers
{
    public const string Unavailable = "Non disponible";
    public const string Breakfast = "Déjeuner";
    public const string Brunch = "Brunch";
    public const string Lunch = "Dîner";
    public const string Supper = "Souper";
}

internal struct Categories
{
    public const string Creative = "Chefs créateurs";
    public const string Pub = "Pubs et microbrasseries";
    public const string Local = "Délices d'ici";
    public const string Restaurant = "Restaurants";
    public const string GoodTables = "Bonnes tables";
    public const string Bistro = "Cafés & Bistros";
    public const string National = "Saveurs du monde";
    public const string Breweries = "Brasseries";
    public const string FamilyOwned = "Cuisine familiale";
    public const string FastFood = "Restauration rapide";
}

File: reservez_web/src/services/FieldHelper.js
export function GetCategories(str){
    let categories = [];
    if (str === null || str.length === 0){
        return categories;
    }
    if (str.length === 1){
        categories.push(GetCategoryString(parseInt(str)));
        return categories;
    }
    const intArray = ParseToIntArray(str);
    return intArray.map(id => GetCategoryString(id));
}

export function GetOffers(str){
    let offers = [];
    if (str === null || str.length === 0 || str === '0'){
        return offers;
    }
    if (str.length === 1){
        offers.push(GetOffersString(parseInt(str)));
        return offers;
    }
    const intArray = ParseToIntArray(str);
    return intArray.map(id => GetOffersString(id));
}

export function FormatPhoneNumber(phoneNb){
    return phoneNb.replace(/(\d{3})(\d{3})(\d{4})/, '($1) $2-$3');
}

function GetCategoryString(id)
{
    switch (id)
    {
        case 1 : return "Chefs créateurs";
        case 2 : return "Pubs et microbrasseries";
        case 3 : return "Délices d'ici";
        case 4 : return "Restaurants";
        case 5 : return "Bonnes tables";
        case 6 : return "Cafés & Bistros";
        case 7 : return "Saveurs du monde";
        case 8 : return "Brasseries";
        case 9 : return "Cuisine familiale";
        case 10 : return "Restauration rapide"
        default : return "Restaurants";
    }
}

function GetOffersString(id)
{
    switch (id)
    {
        case 1 : return "Déjeuner";
        case 2 : return "Brunch";
        case 3 : return "Dîner";
        case 4 : return "Souper";
        default : return "Non disponible";
    }
}

function ParseToIntArray(str){
    return str.split(',').map( Number );
}